app/models/workflow.rb

```diff
@@ -20,7 +19,7 @@ def self.build_given_a_hierarchy(definition)
 
     workflow.first_step = definition[:first_step]
 
-    set_up_the_method(workflow.first_step) { |e| Workflow.build_event_handler_for(workflow.first_step).fire e }
+    set_up_the_method(workflow.first_step) { |e| Workflow.build_event_handler_for(workflow.first_step).receive e }
```

Comment:
you actually dont even need this block here, because it's now the same block as the default block.

Reply:
`set_up_the_method` is probably not needed at all now either, since everything is `receive`

Reply:
well it has some purpose, with the whole 'turn a step into an event handler', but it can be re-examined now

`step[:method]` can be more like `handler.receive` and just be more clear